src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Text } from '../typography/typography';
+
+const Checkbox = (props) => {
+  const [checked, setChecked] = useState(true);
+
+  const handleChange = () => {
+    setChecked(!checked);
+    props.onChange(checked);
+  };
+
+  const HiddenCheckbox = styled.input.attrs({
+    type: 'checkbox',
+    onChange: handleChange,
+  })`
+    height: 1px;
+    width: 1px;
+    margin: -1px;
+    overflow: hidden;
+    position: absolute;
+    visibility: hidden;
+  `;
+
+  const Icon = styled.svg`
+    display: inline-block;
+    fill: none;
+    margin-top: 5px;
+    margin-left: 5px;
+  `;
+
+  const StyledCheckbox = styled.div`
+    display: inline-block;
+    width: 24px;
+    height: 24px;
+
+    background: #f7f8fa;
+    border-radius: 4px;
+
+    ${Icon} {
+      visibility: ${checked ? 'visible' : 'hidden'};
+    }
+  `;
+
+  const CheckboxContainer = styled.div`
+    display: flex;
```

Comment:
`flex` turns this component into a "block element" which means that it will (almost) always be placed on a new line and will be followed by a new line. To make this more flexible we might want to turn this into `inline-flex` which would allow to have a checkbox "inline".
I'm not sure if this is an actual requirement but when using this component I would expect it to be an "inline element".